### DISCUSSION TITLE
nDatabase multi-tenancy support

### DISCUSSION THREAD

**Discussion by chihabhajji** [AliBazzi/IdentityServer4.Contrib.RedisStore#78] [Q&A]
Does the package support the use case of having multiple data stores?

**Comment by AliBazzi** [comment]
can you elaborate more ?

**Reply by chihabhajji** [reply]
I want to implement a SaaS style IAM with the main feature is being able to store and authenticate/authorize users from a remote DB/LDAP

**Reply by AliBazzi** [reply]
Thanks for your response, but you didn't help clarify how that affect Identity Server 4 hosting and the library itself
Identity Server 4 doesn't support multi-tenancy out of the box, you have to work on that and it depends on your styling of the SaaS, is it dedicated instance for each customer or shared ... etc and from there you can see how to customize the library if you follow shared instances.

to summarise, neither identity server 4 nor this library supports multi-tenancy out of the box. you have to work on that.

**Reply by chihabhajji** [reply]
thank you
